src/stepfunctions/steps/states.py

```diff
@@ -218,9 +218,23 @@ def next(self, next_step):
         Returns:
             State or Chain: Next state or chain that will be transitioned to.
         """
-        if self.type in ('Choice', 'Succeed', 'Fail'):
+        if self.type in ('Succeed', 'Fail'):
             raise ValueError('Unexpected State instance `{step}`, State type `{state_type}` does not support method `next`.'.format(step=next_step, state_type=self.type))
 
+        # By design, Choice states do not have the Next field. When used in a chain, the subsequent step becomes the
+        # default choice that executes if none of the specified rules match.
+        # See language spec for more info: https://states-language.net/spec.html#choice-state
+        if self.type is 'Choice':
+            if self.default is not None:
+                logger.warning(
+                    "Chaining Choice state: Overwriting %s's current default_choice (%s) with %s",
```

Comment:
nit: not a blocker but can try using f-string next time: https://www.python.org/dev/peps/pep-0498/ 
Got this recommendation from Shiv and think this could also help here.

Reply:
Good point! Using fstring will make it more readable and faster! Making changes in the next commit!